superset-frontend/src/views/CRUD/hooks.ts

```diff
@@ -678,21 +678,48 @@ export function useDatabaseValidation() {
                         invalid?: string[];
                         missing?: string[];
                         name: string;
+                        catalog: {
+                          name: string;
+                          url: string;
+                          idx: string;
```

Comment:
This should be a number. if `idx` being passed around as a string we should convert it to a number, since that's how we're using it.